In [1]:
import pandas as pd
import os

In [5]:
os.getcwd()

'/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList'

# This is the SemRel target List

In [6]:
current_wd = os.getcwd()
rel_path = "SemRel/targets/random-choice_noun.txt"
abs_path = os.path.join(current_wd, rel_path)

columns = ["Type", "Word", "ID", "Sense", "Sense_Count", "Frequency", "Frequency_Count"]
noun = pd.read_csv(abs_path, sep="\t", header=None, names=columns)

noun.head(40)


Type               Word     ID    Sense  Sense_Count Frequency  \
0    nomen.Artefakt   Anrufbeantworter  11159  senses:            1     freq:   
1    nomen.Artefakt          Schulbuch  15192  senses:            1     freq:   
2    nomen.Artefakt          Atomwaffe  11852  senses:            1     freq:   
3    nomen.Artefakt              Bibel  15132  senses:            1     freq:   
4    nomen.Artefakt        Wandzeitung   9214  senses:            2     freq:   
5    nomen.Artefakt                VHS  12117  senses:            2     freq:   
6    nomen.Artefakt          Bahnsteig   8787  senses:            2     freq:   
7    nomen.Artefakt           Kaufhaus  13510  senses:            2     freq:   
8    nomen.Artefakt             Nutzer  17972  senses:            2     freq:   
9    nomen.Artefakt             Rakete  90623  senses:            2     freq:   
10   nomen.Artefakt             Gatter  14429  senses:            6     freq:   
11   nomen.Artefakt               Bock  11370  senses:            3     freq:   
12   nomen.Artefakt               Trio  16550  senses:            3     freq:   
13   nomen.Artefakt               Kern   8997  senses:            4     freq:   
14   nomen.Artefakt              Gerät   8401  senses:            3     freq:   
15   nomen.Attribut       Lieblichkeit  20054  senses:            1     freq:   
16   nomen.Attribut       Deutlichkeit  19854  senses:            3     freq:   
17   nomen.Attribut           Signatur  18241  senses:            7     freq:   
18     nomen.Besitz         Nobelpreis  20957  senses:            1     freq:   
19     nomen.Besitz             Devise  20285  senses:            2     freq:   
20       nomen.Form             Breite  21852  senses:            1     freq:   
21    nomen.Gefuehl               Ekel  22512  senses:            2     freq:   
22  nomen.Geschehen            Looping  25517  senses:            1     freq:   
23  nomen.Geschehen           Invasion  28672  senses:            1     freq:   
24  nomen.Geschehen      Restaurierung  30239  senses:            1     freq:   
25  nomen.Geschehen               Jagd  23989  senses:            1     freq:   
26  nomen.Geschehen    Gesamtbelastung  29506  senses:            2     freq:   
27  nomen.Geschehen       Bauabschnitt  29380  senses:            2     freq:   
28  nomen.Geschehen            Pension  30012  senses:            2     freq:   
29  nomen.Geschehen        Verhandlung  26753  senses:            2     freq:   
30  nomen.Geschehen        Abschneiden  31003  senses:            5     freq:   
31  nomen.Geschehen         Verjüngung  23625  senses:            3     freq:   
32  nomen.Geschehen              Wache  27360  senses:            4     freq:   
33  nomen.Geschehen          Bedienung  25969  senses:            3     freq:   
34  nomen.Geschehen          Auflösung  30890  senses:            3     freq:   
35     nomen.Gruppe   Materialsammlung  35221  senses:            1     freq:   
36     nomen.Gruppe    Kirchengemeinde  34406  senses:            1     freq:   
37     nomen.Gruppe            Agentur  31387  senses:            1     freq:   
38     nomen.Gruppe  Zivilgesellschaft  31579  senses:            2     freq:   
39     nomen.Gruppe       Mittelklasse  33093  senses:            3     freq:   

    Frequency_Count  
0              1480  
1              3644  
2              5649  
3             30179  
4               428  
5              2479  
6              3064  
7              4616  
8             21899  
9             10253  
10              448  
11             4965  
12             3886  
13            29272  
14            51552  
15              205  
16             2567  
17             6493  
18             3397  
19             5227  
20            13465  
21             2167  
22              257  
23             6001  
24             3307  
25            14624  
26              420  
27             3255  
28             5731  
29            55011  
30         

# Experiment 1 - target list 
Columns:
- target, 
- the response
- the strength of the response (i.e., how many participants generated this relation pair)
- the part-of-speech of the target,
- and the target semantic relation.

In [14]:
path = "SemRel/exp-generation.txt"

columns = ["target", "response", "strength", "part-of-speech of the target", "target semantic relation"]
noun = pd.read_csv(path, sep="\t", header=None, names=columns)

noun[noun['part-of-speech of the target'] == "NOUN"].sort_values(by=['target', 'response'], ascending=True).head(1000)

target     response  strength part-of-speech of the target  \
32    Ablehnung    Abneigung         1                         NOUN   
33    Ablehnung    Abweisung         2                         NOUN   
34    Ablehnung    Akzeptanz         1                         NOUN   
35    Ablehnung  Anerkennung         1                         NOUN   
36    Ablehnung      Annahme         2                         NOUN   
...         ...          ...       ...                          ...   
2698      Katze     Raubtier         1                         NOUN   
2699      Katze    Samtpfote         2                         NOUN   
2701      Katze  Stubentiger         3                         NOUN   
2700      Katze    Säugetier         1                         NOUN   
2702      Katze         Tier         5                         NOUN   

     target semantic relation  
32                        SYN  
33                        SYN  
34                        ANT  
35                        ANT  
36                        ANT  
...                       ...  
2698                      HYP  
2699                      SYN  
2701                      SYN  
2700                      HYP  
2702                      HYP  

[1000 rows x 5 columns]

# Experiment 2 - Rated Word Pair list

-->  Results of Experiment 1 (generation of semantic pairs) and Experiment 2 (rating of strength)

Columns exp-ratings:
- with the target relation, 
- the target word, 
- the generated response, 
- the number of ratings for a strength of 0/1/2/3/4/5, and 
- the number of ratings indicating the number of "I don't know" cases.

In [43]:

# File paths
path1 = "/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList/sem-rel-database/exp-rating/N_ratings_asANT.txt"
path2 = "/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList/sem-rel-database/exp-rating/N_ratings_asHYP.txt"
path3 = "/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList/sem-rel-database/exp-rating/N_ratings_asSYN.txt"

# Read CSV files
ratings1 = pd.read_csv(path1, sep="\t", header=None)
ratings2 = pd.read_csv(path2, sep="\t", header=None)
ratings3 = pd.read_csv(path3, sep="\t", header=None)

# Define column names
columns = ["relation", "target", "response", "N_Strength_0", "N_Strength_1", "N_Strength_2", "N_Strength_3", "N_Strength_4", "N_Strength_5", "N_Idk_cases"]

# Concatenate DataFrames with column names assigned
all_ratings = pd.concat([ratings1, ratings2, ratings3], ignore_index=True)
all_ratings.columns = columns

# Display the first few rows
#all_ratings[all_ratings["relation"] != "ANT"].head(50)


relation            target             response  N_Strength_0  \
574      HYP         Ablehnung            Abneigung             3   
575      HYP         Ablehnung            Abweisung             2   
576      HYP         Ablehnung              Annahme             9   
577      HYP         Ablehnung             Reaktion             0   
578      HYP         Ablehnung            Verhalten             0   
579      HYP         Ablehnung           Zustimmung            10   
580      HYP       Abschneiden            Abtrennen             1   
581      HYP       Abschneiden             Ankleben             9   
582      HYP       Abschneiden              Annähen             9   
583      HYP       Abschneiden             Ergebnis             3   
584      HYP       Abschneiden               Kürzen             1   
585      HYP       Abschneiden               Teilen             0   
586      HYP           Agentur                  Amt             3   
587      HYP           Agentur                Firma             1   
588      HYP           Agentur              Gericht             9   
589      HYP           Agentur      Geschäftsstelle             3   
590      HYP           Agentur          Unternehmen             0   
591      HYP           Agentur  Vermittlungspartner             2   
592      HYP               Alb                  Alm             7   
593      HYP               Alb                 Berg             3   
594      HYP               Alb             Bergdorf             3   
595      HYP               Alb        Gebirgsmassiv             0   
596      HYP               Alb                 Meer            10   
597      HYP               Alb                  Tal             7   
598      HYP  Anrufbeantworter                   AB             5   
599      HYP  Anrufbeantworter       Anrufumleitung             4   
600      HYP  Anrufbeantworter                Gerät             0   
601      HYP  Anrufbeantworter  Kommunikationsgerät             0   
602      HYP  Anrufbeantworter         Standleitung            10   
603      HYP  Anrufbeantworter   Telefonbeantworter             3   
604      HYP              Arzt           Akademiker             1   
605      HYP              Arzt                Beruf             0   
606      HYP              Arzt               Doktor             6   
607      HYP              Arzt           Oberheiler             5   
608      HYP              Arzt              Patient            10   
609      HYP              Arzt           Scharlatan             7   
610      HYP         Atomwaffe           Feuerwaffe             3   
611      HYP         Atomwaffe          Kampfmittel             0   
612      HYP         Atomwaffe            Kernwaffe             0   
613      HYP         Atomwaffe         Nuklearwaffe             4   
614      HYP         Atomwaffe          Vakuumwaffe             6   
615      HYP         Atomwaffe                Waffe             0   
616      HYP         Auflösung           Aufklärung             3   
617      HYP         Auflösung             Beendung             1   
618      HYP         Auflösung                 Ende             1   
619      HYP         Auflösung             Gründung            10   
620      HYP         Auflösung               Rätsel             4   
621      HYP         Auflösung         Verfestigung             9   
622      HYP          Auslöser          Ausschalter             8   
623      HYP          Auslöser              Bremser            10   

     N_Strength_1  N_Strength_2  N_Strength_3  N_Strength_4  N_Strength_5  \
574             0             0             1             3             3   
575             2             0             1             1             4   
576             0             1             0             0             0   
577             0             1             0             3             6   
578             0             0             0             1             9   
579             0     

In [44]:
#Sanity check for relationship types
all_ratings["relation"].value_counts()

relation
ANT    574
HYP    574
SYN    574
Name: count, dtype: int64

In [19]:
len(noun)
noun['part-of-speech of the target'].value_counts()["NOUN"]
noun['strength'].max()


10

# Emotionality and Imaginability List

This list explains the column names


In [50]:
path1 = "/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList/EmoIma/ESM2.csv"
ESM2 = pd.read_csv(path1, sep="," , encoding='latin1')
ESM2.head(50)

Abbreviation  \
0                            IM_Y_M_N\t= Imageability   
1                            IM_Y_F_N\t= Imageability   
2                            IM_O_M_N\t= Imageability   
3                            IM_O_F_N\t= Imageability   
4                         IM_Y_M_MEAN\t= Imageability   
5                         IM_Y_F_MEAN\t= Imageability   
6                         IM_O_M_MEAN\t= Imageability   
7                         IM_O_F_MEAN\t= Imageability   
8                           IM_Y_M_SD\t= Imageability   
9                           IM_Y_F_SD\t= Imageability   
10                          IM_O_M_SD\t= Imageability   
11                          IM_O_F_SD\t= Imageability   
12                           IM_Y_N\t\t= Imageability   
13                           IM_O_N\t\t= Imageability   
14                          IM_Y_MEAN\t= Imageability   
15                          IM_O_MEAN\t= Imageability   
16                          IM_Y_SD\t\t= Imageability   
17                          IM_O_SD\t\t= Imageability   
18                           EM_Y_M_N\t= Emotionality   
19                           EM_Y_F_N\t= Emotionality   
20                           EM_O_M_N\t= Emotionality   
21                           EM_O_F_N\t= Emotionality   
22                        EM_Y_M_MEAN\t= Emotionality   
23                        EM_Y_F_MEAN\t= Emotionality   
24                        EM_O_M_MEAN\t= Emotionality   
25                        EM_O_F_MEAN\t= Emotionality   
26                          EM_Y_M_SD\t= Emotionality   
27                          EM_Y_F_SD\t= Emotionality   
28                          EM_O_M_SD\t= Emotionality   
29                          EM_O_F_SD\t= Emotionality   
30                           EM_Y_N\t\t= Emotionality   
31                           EM_O_N\t\t= Emotionality   
32                          EM_Y_MEAN\t= Emotionality   
33                          EM_O_MEAN\t= Emotionality   
34                          EM_Y_SD\t\t= Emotionality   
35                          EM_O_SD\t\t= Emotionality   
36  F_Unknown_Y\t= Relative frequency of younger a...   
37  F_Unknown_O\t= Relative frequency of older adu...   

                          "metric"      "age group"  "gender"  
0          number of valid ratings   younger adults      male  
1          number of valid ratings   younger adults    female  
2          number of valid ratings     older adults      male  
3          number of valid ratings     older adults    female  
4                   average rating   younger adults      male  
5                   average rating   younger adults    female  
6                   average rating     older adults      male  
7                   average rating     older adults    female  
8    standard deviation of ratings   younger adults      male  
9    standard deviation of ratings   younger adults    female  
10   standard deviation of ratings     older adults      male  
11   standard deviation of ratings     older adults    female  
12         number of valid ratings   younger adults       NaN  
13         number of valid ratings     older adults       NaN  
14                  average rating   younger adults       NaN  
15                  average rating     older adults       NaN  
16   standard deviation of ratings   younger adults       NaN  
17   standard deviation of ratings     older adults       NaN  
18         number of valid ratings   younger adults      male  
19         number of valid ratings   younger adults    female  
20         number of valid ratings     older adults      male  
21         number of valid ratings     older adults    female  
22                  average rating   younger adults      male  
23                  average rating   younger adults    female  
24                  average rating     older adults      male  
25                  average rating     older adults    female  
26   standard deviation of ratings   younger adults      male  
27   standard deviation o

## This list displays the rating results

In [29]:
path = "/Users/Richard/PycharmProjects/NeuroLab_3.10/WordList/EmoIma/ESM1.csv"
Emo = pd.read_csv(path, sep=",")

Emo.head(50)

WORD  IM_Y_M_N  IM_Y_F_N  IM_O_M_N  IM_O_F_N  IM_Y_M_MEAN  \
0          Aal        11        10        12        10        90.55   
1          Aas        10        10        13        11        73.10   
2        Abend        10        11        10        12        53.80   
3    Abendbrot         9        11        13        10        64.89   
4       Abfall         8        10        13         9        73.38   
5      Abgrund        11         9        11        11        72.27   
6       Abhang        11         8        12        11        64.09   
7      Abnahme        11        10        13        10        23.64   
8   Abrechnung        10         9        13        10        47.10   
9        Abruf        14        15        15        17        13.36   
10     Absicht        10         9        12        11        14.40   
11   Absprache        10         8        13         9        27.90   
12  Abstimmung        11        11        13         9        40.91   
13     Abwasch         9         9        13        10        62.78   
14  Abweichung         9        11        12        11        24.44   
15   Abwertung         9        11        12        11        27.56   
16      Achsel        11        10        13        10        72.36   
17  Achterbahn        10         9        12        10        85.60   
18       Acker        10        10        11        12        75.50   
19        Ader        11         9        12        11        86.91   
20       Adler        10         9        11        12        87.90   
21     Admiral        10        10        13         9        62.50   
22        Affe        10        10        14         8        93.80   
23      Affekt         9        12        13        10        14.11   
24       Agent        10         9        11        12        51.00   
25      Ahnung        11         9        12        11        28.82   
26       Ahorn        12         8        13        10        85.67   
27     Akrobat        10         9        13        11        60.20   
28         Akt        10        10        11        12        36.80   
29        Akte        10        10        11        12        67.00   
30      Aktion        11         8        11        12        26.18   
31     Akustik        10        10        13        10        39.20   
32  Albernheit        12        10        12        11        23.00   
33       Album        10        11        13         9        73.10   
34        Alge         8        11        13        10        67.00   
35     Algebra         9        10        12        11        33.89   
36     Alkohol        10        11        12        10        61.80   
37         All        11         9        13        10        71.18   
38       Allee        10         9        11        11        83.60   
39       Alpen        10        11        11        12        87.40   
40       Altar        12         8        10        12        86.67   
41        Alte         9        11        12        11        38.00   
42     Amateur        11         8        12        11        31.09   
43    Ambulanz        10        11        12        11        71.90   
44      Ameise        10        11        12        11        91.80   
45        Amme        10        11        12        11        60.00   
46       Ampel        10        11        12        11        92.80   
47     Amphore        10         8        10        11        65.20   
48       Amsel        10        10        11        12        87.50   
49         Amt        10        10        12        11        42.60   

    IM_Y_F_MEAN  IM_O_M_MEAN  IM_O_F_MEAN  IM_Y_M_SD  ...  EM_O_M_SD  \
0         93.40        97.33        90.50      12.30  ...      27.38   
1         57.50        80.69        59.64      20.44  ...      34.06   
2         52.27        59.90        74.08      28.96  ...      26.29   
3         62.64        84.85        82.50      18.72  ...      22.06   
4         77.80        67.46

In [49]:
len(Emo)

2592

## Merge two lists SemRel and EmoIma 

--> 

In [48]:
#Which item in semrel is also in noun
common_target = all_ratings[all_ratings["target"].isin(Emo["WORD"])]
common_response = noun[noun["response"].isin(Emo["WORD"])]
common_pairs = all_ratings[all_ratings["target"].isin(Emo["WORD"]) & all_ratings["response"].isin(Emo["WORD"])]

# delete duplicate target words (first occurence of the word is kept)
unique_common_pairs = common_pairs.drop_duplicates(subset=['target'])

#unique_common_pairs.head(30)
unique_common_pairs.head(50)


relation     target    response  N_Strength_0  N_Strength_1  N_Strength_2  \
39       ANT       Rübe   Kartoffel             5             1             2   
41       ANT      Fisch     Fahrrad             6             0             0   
51       ANT   Schleife     Viereck             3             0             0   
65       ANT       Funk       Kabel             2             0             1   
83       ANT     Sirene      Stille             1             1             3   
91       ANT      Stuhl       Tisch             3             1             0   
93       ANT      Gerät      Mensch             0             0             1   
112      ANT       Bein         Arm             1             1             0   
120      ANT    Betrieb        Ruhe             1             0             0   
122      ANT     Natter       Vogel             2             0             2   
129      ANT       Hexe         Fee             0             0             1   
132      ANT      Blatt         Ast             1             2             5   
159      ANT      Wache  Einbrecher             1             1             0   
162      ANT    Pfeffer        Salz             3             1             0   
169      ANT     Kadett    Offizier             0             0             3   
170      ANT    Pflanze       Stein             0             0             2   
175      ANT      Zeile       Seite             2             2             2   
182      ANT  Professor     Student             1             0             0   
183      ANT       Zeit        Raum             0             0             3   
185      ANT      Katze        Hund             0             1             0   
194      ANT     Rakete       Bombe             3             2             4   
240      ANT       Arzt       Beruf             5             3             1   
292      ANT      Bauer       Beruf             8             1             1   
324      ANT      Bibel        Buch             5             3             2   
340      ANT       Bock   Säugetier             9             1             0   
349      ANT       Jagd   Tätigkeit             4             4             1   
374      ANT   Kaufhaus    Geschäft             6             3             1   
392      ANT        Los   Schicksal             7             2             1   
464      ANT     Modell    Beispiel             6             2             2   

     N_Strength_3  N_Strength_4  N_Strength_5  N_Idk_cases  
39              2             0             0            0  
41              3             0             1            0  
51              5             2             0            0  
65              1             2             4            0  
83              2             2             1            0  
91              3             1             2            0  
93              2             4             3            0  
112             2             4             2            0  
120             2             6             1            0  
122             3             3             0            0  
129             4             3             2            0  
132             2             0             0            0  
159             1             2             5            0  
162             1             4             1            0  
169             5             0             2            0  
170             3             2             3            0  
175             4             0             0            0  
182             0             4             5            0  
183             2             1             4            0  
185             4             4             1            0  
194             1             0             0            0  
240             1             0             0            0  
292             0             0             0            0  
324             0             0             0            0  
340             0             0             0   

In [47]:
common_pairs["relation"].value_counts()
unique_common_pairs["relation"].value_counts()

relation
ANT    29
Name: count, dtype: int64

In [40]:
print(f'Number of targets in EmoIma: {len(common_target)}')
print(f'Number of responses in EmoIma: {len(common_response)}')
print(f'Number of common Noun-Noun pairs: {len(common_pairs)}')
print(f'Number of common unique Noun-Noun pairs: {len(unique_common_pairs)}')

Number of targets in EmoIma: 519
Number of responses in EmoIma: 461
Number of common Noun-Noun pairs: 204
Number of common unique Noun-Noun pairs: 29


# Summary of List merging Process:

- Number of targets in EmoIma: 519
- Number of responses in EmoIma: 461
- Number of common Noun-Noun pairs: 204
- Number of common unique Noun-Noun pairs: 29

## My Opinion:

Word pair selection based on emotionality and imaginability leads to very few word-pairs (29) At - first glance- no clear semantic relationship emerges for these word pairs.

In [53]:
import os
path = "/Applications/Adobe_Photoshop_2023/Adobe_Photoshop_2023.app"
if os.path.exists(path):
    print("File exists!")
else:
    print("File does not exist.")

File exists!
